public/javascripts/controllers/buyer-ctrl.js

```diff
@@ -227,6 +230,67 @@ angular.module('jasmic.controllers')
                 $scope.demand.de_until = moment().add(7, 'days').toDate();
             };
 
+            $scope.editDemandForm = function (index) {
+
+                $scope.edit_demand = !$scope.edit_demand;
+
+                $scope.demand = $scope.demands[index];
+
+                $scope.demand.de_posting_date= moment().toDate();
+
+                $scope.demand.de_until = moment().add(7, 'days').toDate();
+
+            }
+
+            $scope.editRepForm = function (index) {
+
+                $scope.edit_rep = !$scope.edit_rep;
+
+                $scope.representative = $scope.buyer.re_representatives[index];
+
+            }
+
+            $scope.updateDemand = function() {
+
+                $scope.edit_demand = !$scope.edit_demand;
+
+                DemandEditFactory.update({id:$scope.demands[0].bu_buyer._id, demand_id:$scope.demand._id}, $scope.demand, function(success) {
+
+                    $window.scrollTo(0,0);
+
+                    showDialog($mdDialog, {statusText:"Successfully Updated!"}, false);
+
+                }, function (error) {
+
+                    $window.scrollTo(0,0);
+
+                    showDialog($mdDialog, {statusText:"Error Updating Demand!"}, false);
```

Comment:
The third parameter for `showDialog` is actually a boolean that states whether or not it is an error.  Please adjust to:
```javascript
showDialog($mdDialog, {statusText:"Error Updating Demand!"}, false);
```